Modules/_lsprof.c

```diff
@@ -534,6 +534,7 @@ static int statsForEntry(rotating_node_t *node, void *arg)
 }
 
 /*[clinic input]
+@critical_section self
```

Comment:
I think adding `self` here is unnecessary, same for all others

Reply:
> I think adding `self` here is unnecessary, same for all others

@kumaraditya303 Thank you for the review! Just to confirm, are you suggesting that we should use the lock more granularly and only lock the `ProfilerObject` within specific functions like `ptrace_enter_call()` and `ptrace_leave_call()`?

Reply:
```suggestion
@critical_section
```

No, I meant that just `@critical_section` should be enough here, no need for extra `self`

Reply:
I didn’t know that was valid—I misunderstood your comment earlier. Thank you!